atlasdb-client/src/main/java/com/palantir/atlasdb/keyvalue/impl/DefaultTransactionKeyValueServiceManager.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * (c) Copyright 2024 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.keyvalue.impl;
+
+import com.palantir.atlasdb.cell.api.TransactionKeyValueService;
+import com.palantir.atlasdb.keyvalue.api.KeyValueService;
+import com.palantir.atlasdb.transaction.api.TransactionKeyValueServiceManager;
+import java.util.function.LongSupplier;
+
+public final class DefaultTransactionKeyValueServiceManager implements TransactionKeyValueServiceManager {
+
+    private final KeyValueService delegate;
+    private final TransactionKeyValueService transactionKeyValueService;
+
+    public DefaultTransactionKeyValueServiceManager(KeyValueService delegate) {
+        this.delegate = delegate;
+        this.transactionKeyValueService = new DefaultTransactionKeyValueService(delegate);
+    }
+
+    @Override
+    public TransactionKeyValueService getTransactionKeyValueService(LongSupplier _timestampSupplier) {
+        return transactionKeyValueService;
+    }
+
+    @Override
+    public KeyValueService bestEffortKeyValueService() {
+        return delegate;
+    }
+
+    @Override
+    public void close() {
+        // TODO(jakubk): I don't think this is entirely correct, because we shouldn't own the KeyValueService,
```

Comment:
This is an interesting point: atlasdb dagger does not go through the same initialization and tear-down codepaths that TransactionManagers does.

I think this as a class should not really be responsible for closing the delegate, but I didn't want to mess with this to begin with.

Reply:
We'll likely get to this point when we have a look at KeyValueServiceManager

Reply:
In practice, users of AtlasDB Dagger (the command line tools) as well as the main internal users probably don't worry too much about this.

I think we can use the `Ownable<T>` interface as the input of the KVS and close only if owned?

Reply:
I'll leave as is, we can come back to this.